servicetalk-concurrent-api-internal/src/main/java/io/servicetalk/concurrent/api/internal/SpliceFlatStreamToSingleResult.java

```diff
@@ -44,40 +40,46 @@
 import static java.util.Objects.requireNonNull;
 
 /**
- * This class is responsible for splicing a {@link Publisher}&lt;{@link Object}&gt; with a common {@link Payload}
- * into a {@link Data}&lt;{@link Payload}&gt; eg. {@link StreamingHttpResponse}&lt;{@link Buffer}&gt;.
+ * This operator is responsible for splicing a {@link Publisher}&lt;{@link Object}&gt; into the first emitted element of
+ * type {@link First} followed by a {@link Publisher}&lt;{@link Following}&gt; of remaining elements from the original
+ * {@link Publisher}&lt;{@link Object}&gt;. The spliced stream will be returned as a
+ * {@link Single}&lt;{@link Result}&gt; that should combine {@link First} and {@link Publisher}&lt;{@link Following}&gt;
+ * .
+ * <p>
+ * When {@link Publisher}&lt;{@link Object}&gt; terminates without emitting any item, {@link Single} will complete with
+ * {@code null} instead of {@link Result} and the {@code packer} function won't be invoked.
  *
- * @param <Data> type of container, eg. {@link StreamingHttpResponse}&lt;{@link Buffer}&gt;
- * @param <MetaData> type of meta-data in front of the stream of {@link Payload}, eg. {@link HttpResponseMetaData}
- * @param <Payload> type of payload inside the {@link Data}, eg. {@link Buffer}
+ * @param <First> type of the first element of the {@link Publisher}&lt;{@link Object}&gt;
+ * @param <Following> type of the following elements of the {@link Publisher}&lt;{@link Object}&gt;
+ * @param <Result> type of the result that combines {@link First} and {@link Publisher}&lt;{@link Following}&gt;
  */
-final class SpliceFlatStreamToMetaSingle<Data, MetaData, Payload> implements PublisherToSingleOperator<Object, Data> {
-    private static final Logger LOGGER = LoggerFactory.getLogger(SpliceFlatStreamToMetaSingle.class);
-    private final BiFunction<MetaData, Publisher<Payload>, Data> packer;
+public final class SpliceFlatStreamToSingleResult<First, Following, Result>
```

Comment:
Most of the changes in this file are simple renames. I will highlight the behavior changes